### ISSUE TITLE
ListKeyManager: change emission on QueryList changes

### ISSUE THREAD

**Issue by br-kwon** [angular/components#20978]
https://github.com/angular/components/blob/4699f30f931cc2df2753d2b6e4437ce2b9aafbd1/src/cdk/a11y/key-manager/list-key-manager.ts#L74

In this `if` block, should `this.changes` also emit since it seems to emit indices.

**Comment by crisbeto** [comment]
I'm not sure whether it should emit, because the active item is technically the same, it's just its index that has changed. There's a somewhat-related fix that will emit the event if the active item is removed https://github.com/angular/components/pull/20008.

**Comment by br-kwon** [comment]
Yeah not sure either, but then I feel the `changes` property is quite ambiguous since it emits a number.  I would expect something named `changes` emits when the state of the instance has changed (`activeItemIndex` in this case).

One challenge I faced is when I used this class in an unconventional manner.  Instead of passing in QueryChildren grabbed from ViewChildren or ContentChildren, I passed in a `new QueryList()`. The intention here was to listen for `ListKeyManager.changes()` and `markForCheck()`. When I `reset` the list and `notifyOnChanges`, I would expect an already active item to be "active", but fails to do so in this case.

**Comment by crisbeto** [comment]
That's totally valid, but I suspect that it may cause some breakages. E.g. in `mat-select` we have an event tied to the list manager event which will begin firing if items are added to the select automatically, if we were to make the proposed change. We could try it out once #20008 lands and see how many tests break.

**Comment by br-kwon** [comment]
sounds great! Would love to take a pass at it

**Comment by crisbeto** [comment]
Closing due to inactivity.
